### DISCUSSION TITLE
failed to create bed track

### DISCUSSION THREAD

**Discussion by 1024488648** [GMOD/jbrowse-components#2027] [Q&A] [answered]
When I embedded jbrowse in the web page,failed to create bed track.
Error reporting as'jbrowse-linear-genome-view.js:53 TypeError: Cannot read property '0' of undefined
    at lineCallback (jbrowse-linear-genome-view.js:452)
    at t.<anonymous> (jbrowse-linear-genome-view.js:217)
    at u (jbrowse-linear-genome-view.js:60)
    at Generator._invoke (jbrowse-linear-genome-view.js:60)
    at Generator.next (jbrowse-linear-genome-view.js:60)
    at a (jbrowse-linear-genome-view.js:60)
    at s (jbrowse-linear-genome-view.js:60)'

my profile:
 {
    type: 'BasicTrack',
    trackId:
      'rep1',
    name: 'rep1',
    category: ['samples'],
    assemblyNames: ['Arabidopsis_thaliana'],
    adapter: {
      type: 'BedTabixAdapter',
      bedGzLocation: {
        uri:
          "../../static/data/Arabidopsis_thaliana/bed/1.gz"
      },
      index: {
        location: {
          uri:
            "../../static/data/Arabidopsis_thaliana/bed/1.gz.tbi"
        },
        indexType: 'TBI',
      },
    },
  },

my bed file:
1	33200	33515	ATAC_1_1	316	33300	48	6.47536	2.15794	4.65096
1	37952	38268	ATAC_1_2	317	38075	59	10.96255	2.64237	8.7829
1	55419	55599	ATAC_1_3	181	55423	43	4.76286	1.93774	3.11319
1	56988	57272	ATAC_1_4	285	57190	43	4.76286	1.93774	3.11319
1	88328	88558	ATAC_1_5	231	88474	50	7.22063	2.24602	5.3286

thank you for your answer

**Comment by cmdcolin** [comment, marked as answer]
hi there, this is a little bit due to us being too strict about what a "BED file" is. we could maybe just accept that it is an arbitrary number of fields

you can fix this however if you add a simple "header" line to your bed file naming the fields


so change your file to have a header line like

```
#chr    start   end     name    score   otherfield count     score1     score2      score3
1       33200   33515   ATAC_1_1        316     33300   48      6.47536 2.15794 4.65096
1       37952   38268   ATAC_1_2        317     38075   59      10.96255        2.64237 8.7829
1       55419   55599   ATAC_1_3        181     55423   43      4.76286 1.93774 3.11319
1       56988   57272   ATAC_1_4        285     57190   43      4.76286 1.93774 3.11319
1       88328   88558   ATAC_1_5        231     88474   50      7.22063 2.24602 5.3286
```

that way the data will be correctly displayed in feature detail dialogs also, it wouldn't really know what to name those fields otherwise
